src/main/webapp/resources/js/pages/projects/share/ShareProject.jsx

```diff
@@ -0,0 +1,56 @@
+import { Form, Select, Typography } from "antd";
+import React from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { useGetPotentialProjectsToShareToQuery } from "../../../apis/projects/projects";
+import { setProject } from "./shareSlice";
+
+/**
+ * React component for selecting the project to share a sample with.
+ * @returns {JSX.Element}
+ * @constructor
+ */
+export function ShareProject() {
+  const dispatch = useDispatch();
+  const { currentProject } = useSelector((state) => state.shareReducer);
+  const [options, setOptions] = React.useState();
+
+  /*
+  This fetches a list of the projects that the user has access to.
+   */
+  const {
+    data: projects,
+    isLoading: projectLoading,
+  } = useGetPotentialProjectsToShareToQuery(currentProject, {
+    skip: !currentProject,
+  });
+
+  React.useEffect(() => {
+    if (!projectLoading) {
+      setOptions(
+        projects.map((project) => ({
+          label: project.name,
+          value: project.identifier,
+        }))
+      );
+    }
+  }, [projects, projectLoading]);
+
+  return (
+    <Form.Item
+      label={
+        <Typography.Text strong>
+          {i18n("ShareSamples.projects")}
+        </Typography.Text>
+      }
+    >
+      <Select
```

Comment:
Rather than just a `Select` for the project listing here is there a way we can add a search?  Some users may have many projects and the simple select might not be the easiest to use.